maildir/shared.c

```diff
@@ -1139,7 +1139,14 @@ void maildir_canon_filename(struct Buffer *dest, const char *src)
     src = t + 1;
 
   mutt_buffer_strcpy(dest, src);
-  char *u = strrchr(dest->data, ':');
+  char *u = strchr(dest->data, ',');
+  if (u)
+  {
+    *u = '\0';
+    dest->dptr = u;
+    return;
+  }
+  u = strrchr(dest->data, ':');
   if (u)
```

Comment:
I guess if we found ',' and changed that into a '\0' we won't find ':' anymore. Can we add a comment saying how we expect the format to be and how we want to modify it?

Reply:
Correct. The "," is used at least by mbsync and dovecot (probably more) to add some additional attributes *before* the flags. The old code only strips the flags. Since we know that we can already stop when we find a ",".
I can add comment to the commit, no problem.

Reply:
Thanks!